examples/nginx-grafana/README.md

```diff
@@ -0,0 +1,340 @@
+* [`01_nginx-access-log.sql`](01_nginx-access-log.sql). 
+
+The SQL file contains the following DDL:
+![SQL code screenshot with line numbers](images/03_nginx-access-log.png)
+
+### Generating Data to Mimic Nginx Access Log Data
+On line 3, the [`CREATE RANDOM STREAM`](https://docs.timeplus.com/proton-create-stream#create-random-stream) DDL is used to create a random stream named `nginx_access_log`. 
+
+Each column is randomly assigned a `default` value as you can see on lines 4 - 30.
+
+The table below goes over each of the 12 columns and the database [functions](https://docs.timeplus.com/functions) used to generate test data.
+| #  | Column | Generated Data | 
+|------|--------|------------------|
+|1 | `remote_ip` | [`random_in_type('ipv4')`](https://docs.timeplus.com/functions_for_random#random_in_type): returns a random [ipv4](https://docs.timeplus.com/datatypes) representing a user's IP address. |
+| 2 | `rfc1413_ident` | `default`s to '-' but should be a string conforming to [RFC1413](https://datatracker.ietf.org/doc/html/rfc1413). Currently unused. |
+| 3 | `remote_user` | `default`s to '-' since most blog traffic is from unauthenticated users. Currently unused. |
+| 4 | `date_time` | `random_in_type('datetime64', 365, y -> to_time('2023-6-17') + interval y day)`: [`random_in_type('datetime64')`](https://docs.timeplus.com/functions_for_random#random_in_type) between 2023-6-17 & a 1-yr interval i.e. [2023-06-17, 2024-06-17). |
+| 5 | `http_method` | `['GET', 'POST', 'PUT', 'DELETE', 'HEAD'][rand()%5]`: uses [`rand()`](https://docs.timeplus.com/functions_for_random#rand) to return a random index between [0, 5) in this 5-element array. The array samples 5 of the [39 HTTP verbs](https://stackoverflow.com/questions/41411152/how-many-http-verbs-are-there). |
+| 6 | `path` | `['/rss/', '/', '/sitemap.xml', '/favicon.ico', '/robots.txt', ...][rand()%11]`: uses [`rand()`](https://docs.timeplus.com/functions_for_random#rand) to return a random index between [0, 11) in this 11-element array of sample URL subpaths. |
+| 7 | `http_version` | `['HTTP/1.0', 'HTTP/1.1', 'HTTP/2.0'][rand()%3]`: similar to `path`. |
+| 8 | `status` | `[200, 301, 302, 304, 400, 404][rand()%6]`: similar to `path`. |
+| 9 | `size` | `rand()%5000000`: uses `rand()` to generate a random amount of [bytes sent](https://stackoverflow.com/a/30837653), up to `~5MB`. |
+| 10 | `referer` | `['-', 'https://ayewo.com/', '...', 'https://google.com/'][rand()%4]`: similar to `path`. |
+| 11 | `user_agent` | `['...', '...', ...][rand()%14]`: similar to `path`. |
+| 12 | `malicious_request` | `if(rand()%100 < 5, '\x16\...', '')`: returns a malformed sequence whenever `rand()%100` is < 5. |
+  
+
+
+### Random Streams
+There is an important note about the design of random streams in Timeplus Proton mentioned in the [`RANDOM STREAM`](https://docs.timeplus.com/proton-create-stream#create-random-stream) documentation that is worth highlighting here:
+> [!NOTE]
```

Comment:
I realized there could be a misunderstanding based on my previous doc, that people may think all data generated will be kept in the memory. I changed that doc to the following text. Please quote it or highlight it in your own words

When you run a Timeplus SQL query with a random stream, the data will be generated and analayzed by the query engine. Depending on the query, all generated data or aggregated states can be kept in memory during the query time. If you are not querying the random stream, there is no data generated or kept in memory.

Reply:
It's still there. Perhaps you updated a different page?

On https://docs.timeplus.com/proton-create-stream#create-random-stream when I do a Ctrl+F for this text: 
> The data of random stream is kept in memory during the query time. If you are not querying the random stream, there is no data generated or kept in memory.

Reply:
I will publish that doc change in 1 or 2 hours. Same link